### DISCUSSION TITLE
Final Project

### DISCUSSION THREAD

**Discussion by kassyoveson** [USU-ITLS-x265-23-summer/class-repo#242] [General]
As fun as this class has been, I am happy that its coming to a close. My project is not very spectacular compared to some others. I set out to create a basic form that schools can use to take a daily lunch count. This helps the cafeteria mangers collect data quickly and keep good records for audit season. This is the link. Enjoy the rest of your summer!

https://usu-itls-x265-23-summer.github.io/Kassy-Oveson--Project/

**Comment by coopsand** [comment]
Do not discount yourself! You worked long and hard and your site is incredibly useful!! You can always continue to learn more after the course is over as well! I think it looks great! I'm sure this form would be a huge help in cafeterias. I love the color scheme you chose as well.

**Comment by bst3ck** [comment]
I think you did great. Well done. You would be surprised how many sites are not elaborate but instead are simple pages meant for individual purposes. And your page accomplished that task. Great job.

**Comment by jakegros** [comment]
It is straightforward and useful.  A lot of times extremely elaborate sites may look "cool and fancy" but do not function properly or help.  This functions.  I like how the date is filled in automatically. Nice work.  Does the submit button give you a confirmation?  I don't know if it was just giving me an issue.

**Reply by kassyoveson** [reply]
I didn't code a submission page since I couldn't connect it to the server, and I don't know how to create a database to collect the data. I may extend this project to actually work for my capstone project.

**Reply by jakegros** [reply]
Yes, that is a lot extra to do.  I just meant giving a message that the form has been "submitted" (just the message even without it going anywhere).
I added a js script just for my form:

document.getElementById("contact-form").addEventListener("submit", function (event) {
    event.preventDefault();

    const formData = new FormData(event.target);
    const data = {};
    formData.forEach((value, key) => {
        data[key] = value;
    });

    // This is a simulated form submission
    // Replace this with actual form submission to a backend endpoint using Fetch API
    setTimeout(() => {
        document.getElementById("status").textContent = "Form submitted successfully!";
        document.getElementById("contact-form").reset();
    }, 1000);
));

**Reply by kassyoveson** [reply]
Thanks. That is very helpful!

**Comment by gabebott** [comment]
This project is great! I don't think this is underwhelming at all. This is a very functional and useful project. I think that you created something that people can use, and that will make their lives easier. Having a tool like this can be a big plus for people to immediately have a record of something that usually takes more effort than an online survey. I think you did a really good job and I think that the site is very easy to navigate.Great job!
